lib/tapioca/dsl/compilers/url_helpers.rb

```diff
@@ -136,15 +145,16 @@ def gather_non_discoverable_includers
 
           sig { params(mod: Module, helper: Module).returns(T::Boolean) }
           private def includes_helper?(mod, helper)
-            superclass_ancestors = []
+            return ancestors_of(mod).include?(helper) unless Class === mod
+
+            superclass = superclass_of(mod)
 
-            if Class === mod
-              superclass = superclass_of(mod)
-              superclass_ancestors = ancestors_of(superclass) if superclass
+            ancestors_of(mod).each do |a|
+              return true if helper.equal?(a)
+              return false if superclass.equal?(a)
             end
 
-            ancestors = Set.new.compare_by_identity.merge(ancestors_of(mod)).subtract(superclass_ancestors)
-            ancestors.any? { |ancestor| helper == ancestor }
```

Comment:
This block was another culprit in the hot path. `include?` was faster.

Reply:
Nevermind, this breaks in the XPath case :')

https://github.com/Shopify/tapioca/pull/677